### PR TITLE
Added deleteItem and deleteItems methods

### PR DESCRIPTION
**Pull request by glecko**

### CODE REVIEW

**Comment by glecko** [comment]
First approach I tried the function to take an item as an argument (similar to update).

However, I found it was a kind of deceiving approach, since at the moment that you delete an item, all the indexes of the items the user may have stored in memory outside of Table become invalid.

Example:
    var table = Sheetfu.getTable(sheetName, headerRow);       
    var item = table.select(...).first(); // Has index 2
    var item2 = table.select(...).first(); // Has index 7
    table.deleteOne(item);
    // At this point, inside table, item2 has the index 6. However, the object assigned to "item2" still has index 7. If we now do table.deleteOne(item2) we are not actually deleting the original item2, but another random item that is just below item2.

If we make it so you have to specify the index, it's much more intuitive what will be deleted.

**Comment by glecko** [comment]
Another thing I just thought of that I hadn't though of before:

With this approach, deleting should not lead to unexpected results, but deleting and then updating does! The index of the item in the user's variable is still wrong, and it's used by the Table.update method, resulting in overwriting a random item instead of the correct one.

So either we make updating also index-based, or we have to think of a completely new way of approaching this...

File: table.gs [hunk 1]
```diff
@@ -286,6 +286,28 @@ Table.prototype.updateMany = function(manyItems) {
   }
 };
 
+
+/**
+ * Method to delete one item from the items grid.
+ * @param {index} int: The index at which to delete an item
```
Comment (urpizu) [inline table.gs]: It is not an int, it is a Item object

**Review by urpizu — APPROVED** [review]

**Comment by philippe2803** [comment]
Really cool stuff. 

I think we should consider the deleteMany method for this PR. Otherwise, we could have users using the deleteOne method when for-looping the items. This could lead to unexpected results.

**Review by philippe2803 — CHANGES_REQUESTED** [review]
I approve the current code. We should add the deleteMany method in it.

**Comment by glecko** [comment]
Not sure the deleteMany function is entirely needed. Right now it is possible to delete items in a for loop through a selection:

```
var selection = table.selection(...);
for(var i = 0; i < selection.length; i++)
{
   table.deleteOne(selection[i]);
}
```

What won't work is iterating through table.items and deleting in a loop (externally keeping track of indexing). But the only scenario I can think of a user would do that is when he wishes to delete all items in the table (for which deleteAll() already exists).

It's true though that deleteMany will make user code cleaner; and we can fine-tune its performance to be a bit better than just calling deleteOne in a loop.

**Comment by urpizu** [comment]
Hmm... Even I was with the same thoughts as you @glecko, @philippe2803 convinced me!

One user could use a table loop with an if condition inside, which make sense and it could lead to unexpected results. I'd add the deleteItems(criteria) as it's the proper way, and not evident if you don't know the library properly!

**Comment by glecko** [comment]
Added the deleteItems() method based on selection criteria

**Review by urpizu — APPROVED** [review]
Great job! (Y)

**Comment by philippe2803** [comment]
This is not what I had in mind for the `deleteMany` methods. I was under the impression that it would take an array of items as input, instead of filters. This is actually a really cool feature on its own though so we should definitely keep it.

I think this is worthwhile however to have a method that delete multiple items using an array of items as an input, mainly because we could have conditions based on something that is not values (e.g. if cell colour background is red), which the select method only deal with.

From a documentation standpoint, we need to explain them as follow too. 
1 - This is how to delete one item.
2 - This is how to delete multiple items (first find every items you want to delete, put them in an array, then use the deleteMany method).
3 - This is how to remove every values matching some filters. 

The current deleteItems is really good though so let's keep it. From a naming convention, I have a personal preference for the following:
 - `deleteOne`: delete one item.
 - `deleteMany`: delete every items for a given array.
 - `remove`:  when using filters for values.

I feel `deleteItem` and `deleteItems` are a bit too similar.

Thoughts?

**Comment by glecko** [comment]
Completely agree. Only thing is, we cannot use "remove" as a function name as it already a native javascript function name.

What about "deleteByFilter" or "deleteByCriteria"?

**Comment by sp-guillem-orpinell** [comment]
I'd use deleteByCriteria(filterObject) ;)

**Comment by glecko** [comment]
Or perhaps deleteSelection() ? :D

**Review by sp-guillem-orpinell — APPROVED** [review]
Looks fine to me!

**Review by philippe2803 — APPROVED** [review]

**Pull request merged**